### DISCUSSION TITLE
Problem Load custom .stl file in Ultrasound Simulator

### DISCUSSION THREAD

**Discussion by lucabror** [PlusToolkit/PlusLib#942] [Q&A]
Hi all,
I've a problem with loading a custom .stl file in Ultrasound Simulator.
I made a generic geometry to test custom stl model. I loaded it in Slicer3D I hardened the transform in Slicer inside the gel block and exported it to STL in my custom position.
When I start simulation through PlusServer.exe I can't see my custom geometry (stl file) but I can see just a echo signal of cilinder but not the echo signal of my custom model.

Here is a screenshot. Why my custom model doesn’t appears in echo images in red Windows?
Thank you
Best 
Luca
![image](https://user-images.githubusercontent.com/33671113/166410709-1aaadd33-922f-46a3-9a3f-06599d0a5607.png)

**Comment by lassoan** [comment]
In recent Slicer versions, models are saved by default in LPS coordinate system. If the file was in RAS coordinate system (or the coordinate system was manually changed in the storage node before saving) then the file is saved in RAS coordinate system, but otherwise it is saved in LPS. The coordinate system is stored in the file header, so Slicer always know how to load the model correctly (Slicer uses RAS coordinate system, so if it loads an model in LPS coordinate system then it flips the sign of the first two coordinate values).

In Plus, the coordinate system information is not read from the header (this could be a feature implemented later), so you need to make sure that the coordinates stored in the file are consistent with `ObjectCoordinateFrame` that you specify. Most likely your problem is that Slicer saves the model in LPS coordinate system but your Plus configuration file is set up so that you expect the model in RAS coordinate system instead.

Potential solutions:

Solution A: Change how you save from Slicer, by calling `modelNode.GetStorageNode().SetCoordinateSystem(slicer.vtkMRMLStorageNode.CoordinateSystemRAS)` before you save the model node.

Solution B:. Change the `ObjectCoordinateFrame` in the Plus configuration file. For example, if you have `ObjectCoordinateFrame="Reference"` in the configuration file and your `Reference` coordinate system is RAS then you can add a `ReferenceLPS` coordinate system, specify the `Reference` to `ReferenceLPS` transform (diagonal matrix of `<-1,-1,1,1>) and use `ObjectCoordinateFrame="ReferenceLPS"`.
